app/notification.go

```diff
@@ -613,6 +613,16 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 					}
 					message.Add("thread", string(payload))
 
+					previousUnreadMentions := userThread.UnreadMentions
+					previousUnreadReplies := userThread.UnreadReplies - 1
+
+					if mentions.isUserMentioned(uid) {
+						previousUnreadMentions = userThread.UnreadMentions - 1
+					}
+
+					message.Add("previous_unread_mentions", previousUnreadMentions)
+					message.Add("previous_unread_replies", previousUnreadReplies)
```

Comment:
Can we add an app layer test for this?

Reply:
@agnivade can you spy on a method call in go? I'd appreciate some pointers on how to proceed testing this